File: 125.valid-palindrome/solution.ts
function isPalindrome(s: string): boolean {
  let left = 0;
  let right = s.length - 1;
  s = s.toLowerCase();
  const alphabets = new Set('abcdefghijklmnopqrstuvwxyz0123456789'.split(''));
  while (left <= right) {
      while (left < s.length && !alphabets.has(s[left])) { 
          left++;
      }
      while (0 <= right && !alphabets.has(s[right])) { 
          right--;
      }
      if (s[left] !== s[right]) {
          return false;
      }
      left++;
      right--;
  }
  return true;
};

File: 125.valid-palindrome/solution.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        left = 0
        right = len(s) - 1
        s = s.lower()
        while (left < right):
            while left < right and not s[left].isalnum():
                left += 1
            while left < right and not s[right].isalnum():
                right -= 1
            if s[left] != s[right]:
                return False
            left += 1
            right -= 1
        return True